From f0b37d5b7902a56e73ec0372ec000ae474a377b7 Mon Sep 17 00:00:00 2001
From: Jimmy Chen <jimmycmchen@google.com>
Date: Thu, 22 Sep 2022 13:50:09 +0800
Subject: [PATCH] [DO NOT MERGE] passpoint: validate decorated identity prefix

Bug: 246539931
Test: atest FrameworksWifiTests
      insatll PoC to verify the issue does not happen
Change-Id: I59144c43d2ee9590c8d973e24877e56aaf395b33
(cherry picked from commit df9d477011d9e64857dea3658cba1fc25e5d18cc)
(cherry picked from commit 2e0196f10caaba981ec595d4c67a820eb802f74b)
Merged-In: I59144c43d2ee9590c8d973e24877e56aaf395b33
---
 .../android/net/wifi/hotspot2/PasspointConfiguration.java | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/framework/java/android/net/wifi/hotspot2/PasspointConfiguration.java b/framework/java/android/net/wifi/hotspot2/PasspointConfiguration.java
index 375c9fb4c..239f406ac 100644
--- a/framework/java/android/net/wifi/hotspot2/PasspointConfiguration.java
+++ b/framework/java/android/net/wifi/hotspot2/PasspointConfiguration.java
@@ -37,6 +37,7 @@ import android.telephony.SubscriptionInfo;
 import android.telephony.SubscriptionManager;
 import android.telephony.TelephonyManager;
 import android.text.TextUtils;
+import android.util.EventLog;
 import android.util.Log;
 
 import androidx.annotation.RequiresApi;
@@ -952,6 +953,13 @@ public final class PasspointConfiguration implements Parcelable {
         if (mPolicy != null && !mPolicy.validate()) {
             return false;
         }
+        // Optional: DecoratedIdentityPrefix
+        if (!TextUtils.isEmpty(mDecoratedIdentityPrefix)
+                && !mDecoratedIdentityPrefix.endsWith("!")) {
+            EventLog.writeEvent(0x534e4554, "246539931", -1,
+                    "Invalid decorated identity prefix");
+            return false;
+        }
 
         if (mTrustRootCertList != null) {
             for (Map.Entry<String, byte[]> entry : mTrustRootCertList.entrySet()) {
-- 
2.38.1.273.g43a17bfeac-goog

